messaging/src/main/java/org/axonframework/messaging/annotation/HandlerAttributes.java

```diff
@@ -0,0 +1,144 @@
+     * @param handlerType the type of handler to add {@code attributes} for
+     * @param attributes  the {@link Map} of attribute information to add for the given {@code handlerType}
+     */
+    public void put(String handlerType, Map<String, Object> attributes) {
+        this.attributes.put(handlerType, attributes);
+    }
+
+    /**
+     * Retrieve the attributes for the given {@code handlerType}. Might be {@code null} if there are no attributes
+     * present for the specified type.
+     *
+     * @param handlerType the type of handler to retrieve attributes for
+     * @return the attributes for the given {@code handlerType}
+     */
+    public Map<String, Object> get(String handlerType) {
+        return Collections.unmodifiableMap(attributes.get(handlerType));
+    }
+
+    /**
+     * Retrieve all attributes for all handler types stored in this {@link HandlerAttributes} object. The returned
+     * {@link Map} contains a {@link Map} of attribute name to attribute value, per handler type present.
+     *
+     * @return all attributes for all handler types stored in this {@link HandlerAttributes} object
+     */
+    public Map<String, Map<String, Object>> getAll() {
+        return Collections.unmodifiableMap(
+                attributes.entrySet().stream()
+                          .collect(Collectors.toMap(
+                                  Map.Entry::getKey,
+                                  entry -> Collections.unmodifiableMap(entry.getValue())
+                          ))
+        );
+    }
+
+    /**
+     * Retrieve all attributes for all handler types stored in this {@link HandlerAttributes} object in a prefixed
+     * format. The returned {@link Map} contains keys representing a format of {@code "[handlerType].[attributeName]"}
+     * with the respective attribute value corresponding to the attribute name.
+     * <p>
+     * Serves the purpose a of providing a {@code Map} containing a property file like structure.
+     *
+     * @return all attributes for all handler types stored in this {@link HandlerAttributes} object in a prefixed format
+     */
+    public Map<String, Object> getAllPrefixed() {
+        Map<String, Object> allPrefixedAttributes = new HashMap<>();
+        for (Map.Entry<String, Map<String, Object>> attributeEntry : attributes.entrySet()) {
+            String handlerType = attributeEntry.getKey();
+            Map<String, Object> prefixedAttributes =
+                    attributeEntry.getValue().entrySet().stream()
+                                  .collect(Collectors.toMap(
+                                          entry -> prefixedKey(handlerType, entry.getKey()),
+                                          Map.Entry::getValue
+                                  ));
+            allPrefixedAttributes.putAll(prefixedAttributes);
+        }
+        return Collections.unmodifiableMap(allPrefixedAttributes);
+    }
+
+    private String prefixedKey(String handlerType, String attributeName) {
```

Comment:
We should abstract this logic away into whatever component inspects the attributes. This still seems quite specific to the annotation approach.

Reply:
I've taken your idea of constructing an `AnnotatedHandlerAttributes` class. It's this object which currently contains all this logic, so essentially it's abstracted away to be annotation specific right now.